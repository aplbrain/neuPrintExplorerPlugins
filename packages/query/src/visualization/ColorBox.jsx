import React from 'react';
import PropTypes from 'prop-types';

function ColorBox({ margin, width, height, backgroundColor, title, text, color='#000' }) {
  const styles = {
    margin: `${margin}px`,
    width: `${width}px`,
    minWidth: `${width}px`,
    height: `${height}px`,
    minHeight: `${height}px`,
    backgroundColor,
    color,
    overflow: 'visible',
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  };
  return (
    <div style={styles} title={title}>
      {text}
    </div>
  );
}

ColorBox.propTypes = {
  margin: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired
};

export default ColorBox;
