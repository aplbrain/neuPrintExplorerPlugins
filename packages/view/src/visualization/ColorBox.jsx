import React from 'react';
import PropTypes from 'prop-types';

function ColorBox({ margin, width, height, backgroundColor, title, text }) {
  const styles = {
    margin: `${margin}px`,
    width: `${width}px`,
    minWidth: `${width}px`,
    height: `${height}px`,
    minHeight: `${height}px`,
    backgroundColor,
    overflow: 'visible',
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row'
  };
  return (
    <div style={styles} title={title}>
      {text}
    </div>
  );
};

ColorBox.propTypes = {
  margin: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.string,
  ]).isRequired,
};

export default ColorBox;
