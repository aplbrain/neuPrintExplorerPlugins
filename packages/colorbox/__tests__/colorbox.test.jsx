import React from 'react';
import renderer from 'react-test-renderer';
import ColorBox from '../src/colorbox';

test('ColorBox uses correct color', () => {
  const raw = <ColorBox margin={0} width={10} height={20} backgroundColor='#fff'
        key='foo'
        title='100%'
        text="alt text"
      />

  const component = renderer.create(raw);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
