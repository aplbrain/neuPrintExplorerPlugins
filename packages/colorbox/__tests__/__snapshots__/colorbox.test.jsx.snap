// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox uses correct color 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#fff",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "color": "#000",
      "display": "inline-flex",
      "flexDirection": "column",
      "height": "20px",
      "justifyContent": "center",
      "margin": "0px",
      "minHeight": "20px",
      "minWidth": "10px",
      "overflow": "visible",
      "width": "10px",
    }
  }
  title="100%"
>
  alt text
</div>
`;
